presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,150 @@
+import org.testng.annotations.Test;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.tests.TestGroups.CANCEL_QUERY;
+import static com.google.common.base.Preconditions.checkState;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.lang.System.nanoTime;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static org.testng.Assert.fail;
+
+public class SqlCancelTests
+        extends ProductTest
+{
+    private ExecutorService executor;
+    private QueryCanceller queryCanceller;
+
+    @Inject
+    @Named("databases.presto.server_address")
+    private String serverAddress;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        executor = newFixedThreadPool(2);
+        queryCanceller = new QueryCanceller(serverAddress);
+    }
+
+    @AfterTestWithContext
+    public void cleanUp()
+    {
+        if (executor != null) {
+            executor.shutdownNow();
+        }
+        if (queryCanceller != null) {
+            queryCanceller.close();
+        }
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60 * 1000L)
+    public void cancelCreateTableTest()
+            throws Exception
+    {
+        String tableName = format("cancel_createtable_%s", nanoTime());
+        String sql = format("CREATE TABLE %s AS SELECT * FROM tpch.sf1.lineitem", tableName);
+
+        runAndThenCancelQuery(sql);
+        assertThat(() -> query(format("SELECT * from %s", tableName)))
+                .failsWithMessage(format("Table hive.default.%s does not exist", tableName));
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60 * 1000L)
+    public void cancelInsertIntoTest()
+            throws Exception
+    {
+        String tableName = format("cancel_insertinto_%s", nanoTime());
+        query(format("CREATE TABLE %s " +
+                "(orderkey BIGINT, partkey BIGINT, shipinstruct VARCHAR(25)) ", tableName));
+        String sql = format("INSERT INTO %s " +
+                "SELECT orderkey, partkey, shipinstruct FROM tpch.sf1.lineitem", tableName);
+
+        runAndThenCancelQuery(sql);
+        assertThat(query(format("SELECT * from %s", tableName))).hasNoRows();
+        query(format("DROP TABLE %s", tableName));
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60 * 1000L)
+    public void cancelSelectTest()
+            throws Exception
+    {
+        runAndThenCancelQuery(format("SELECT * FROM tpch.sf1.lineitem AS cancel_select_%s", nanoTime()));
+    }
+
+    private void runAndThenCancelQuery(String sql)
+            throws Exception
+    {
+        Future<?> executeQueryThread = executor.submit(() -> assertThat(() -> query(sql)).failsWithMessage("Query was canceled"));
```

Comment:
It is not clear what happens if  this assertion fails.
I would rather have here just `executor.submit(() -> query(sql))` and check exception message in the main test thread.